src/cc-index.ts

```diff
@@ -0,0 +1,185 @@
+import {Downloader} from "./downloader";
+
+/**
+ * Objects of this class represent responses from the CommonCrawl index.
+ */
+export class CCIndexResponse {
+
+    public urlkey : string;
+    public timestamp : string;
+    public status : number;
+    public url : string;
+    public filename : string;
+    public length : number;
+    public mime : string;
+    public offset : number;
+    public digest : string;
+
+
+}
+
+/**
+ * This class allows interaction with the CommonCrawl index to search for specific URLs.
+ */
+export class CCIndex {
+
+    public static defaultCCIndex = "http://index.commoncrawl.org/CC-MAIN-2017-04-index";
+
+
+    /**
+     * Main function to interact with the CommonCrawl index.
+     * Takes a URL to look up, queries the CC index, parses the CC response and returns
+     * an array of relative paths to all WET files tha should include the URL.
+     *
+     * Either an error or at least one path are passed to the callback. If no WET files were found, an error is passed.
+     *
+     * @param lookupURL             URL to look up in the CC index
+     * @param callback              will be called with an array of paths to WET files (set of strings)
+     * @param ccIndexPageURL        (optional) base URL of the CC index page; if not provided -> defaultCCIndex is used
+     */
+    public static getWETPathsForURL(lookupURL : string,
+                                    callback : (err? : Error, wetPaths? : Array<string>) => void,
+                                    ccIndexPageURL? : string ) {
+        CCIndex.lookUpURL(lookupURL, (err, rawResponse) => {
+            if (err) {  callback(err);       return; }
+
+            // got a raw response -> parse it
+            let resObjs = CCIndex.parseStringToCCIndexResponse(rawResponse);
+            // and create WET paths
+            let wetPaths = CCIndex.constructWETPaths(lookupURL, resObjs);
+
+            if (wetPaths.length < 1) {
+                callback(new Error("No WET files found for " + lookupURL));
+            } else {
+                callback(undefined, wetPaths);
+            }
+
+
+        }, ccIndexPageURL);
+
+    }
+
+
+    /**
+     * [HELPER FUNCTION]
+     * Query the CC index with a URL and call the callback with the RAW response string.
+     *
+     * @param lookupURL             URL to look up in the CC index
+     * @param callback              (optional) will be called with the response body string; if not provided -> output to console
+     * @param ccIndexPageURL        (optional) base URL of the CC index page; if not provided -> defaultCCIndex is used
+     */
+    public static lookUpURL(lookupURL : string,
```

Comment:
Set the method to `private` if it's only a helper function? That way we won't get confused what the main method is when using this class.